compiler/luci/service/src/CircleTypeInferenceRule.cpp

```diff
@@ -93,6 +93,8 @@ struct TypeInferenceAlgorithm final : public luci::CircleNodeVisitor<loco::DataT
 
   loco::DataType visit(const luci::CircleCos *node) final { return luci::dtype_get(node->x()); }
 
+  loco::DataType visit(const luci::CircleCumSum *node) final { return node->dtype(); }
```

Comment:
Maybe? Because CumSum has same dtype with its input.
```suggestion
  loco::DataType visit(const luci::CircleCumSum *node) final { return luci::dtype_get(node->input()); }
```

Reply:
Both of them have same type, but it's better to following dtype of input as you commented. :)
Thanks :smile:

Reply:
> Both of them have same type

To be exact, `node-dtype()` is from the file dtype.
`luci::dtype_get(node->input())` is setting dtype from the input.
Logically, I think the second one is better as we're doing dtype inference.
As I understand `CumSum` output dtype follows input dtype.